### DISCUSSION TITLE
queryRaw w/ postgres function throws invalid invocation but hardcoded query does not.

### DISCUSSION THREAD

**Discussion by SCD-24** [prisma/prisma#10476] [Q&A] [answered]
Hi everyone.  I'm trying to use $queryRaw that uses a function I made in postgres

```
const result = await prisma.$queryRaw\`SELECT * from job_search_db.jobs_function(
		${distanceInt}, 
		${dateRangeStr}, 
		${remoteBoolArr}, 
		${citiesArr}, 
		${termsArr}, 
		${excludeTermsArr}, 
		${excludeCompaniesArr}, 
		${jobBoardsArr}, 
		${orderBy}, 
		${limitInt}, 
		${offsetInt}
	)\`
```
when I format it like this I get: 

```
PrismaClientKnownRequestError: 
Invalid `prisma.queryRaw()` invocation:


  Raw query failed. Code: `54000`. Message: `db error: ERROR: number of array dimensions (1634890337) exceeds the maximum allowed (6)`
    at cb (/search-backend/prisma/generated/output/runtime/index.js:38675:17)
    at async resolve (/search-backend/dist/types/queries/jobs.js:235:28)
    at async middleware (search-backend/node_modules/graphql-shield/dist/generator.js:30:24) {
  code: 'P2010',
  clientVersion: '3.5.0',
  meta: {
    code: '54000',
    message: 'db error: ERROR: number of array dimensions (1634890337) exceeds the maximum allowed (6)'
  }
}
```

However, when I just log the query that I placed in a variable then paste the query in hardcoded it works and returns the expected results

```
const result = await prisma.$queryRaw\`SELECT * from job_search_db.jobs_function(
		50000, 
		'2021-09-29T17:24:58.142Z', 
		array[true, false], 
		array[4560349,5128581,5205659], 
		array['%javascript%','%react.js%'], 
		array['%junior%','%jr%','%manager%','%analyst%','%recruiter%','%accounting%','%network%','%associate%'], 
		array['cybercoders'], 
		null, 
		'desc', 
		26, 
		0
)\`
```

Does anyone have any ideas as to why this is?

**Comment by SCD-24** [comment, marked as answer]
Solved:  I cast to the appropriate types. that gave me an error

`'incorrect binary data format in bind parameter 1' `

After digging a bit I came across this issue: [$queryRaw doesn't allow for typecasts](https://github.com/prisma/prisma/issues/9197)
Using $queryRawUnsafe() solved the issue for now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
